src/sagemaker/tensorflow/deploying_python.rst

```diff
@@ -0,0 +1,190 @@
+  - start a Python-based HTTP server which supports protobuf, JSON and CSV content types, and can run your custom
+    input and output python functions. See `Making predictions against a SageMaker Endpoint`_.
+
+
+When the ``deploy`` call finishes, the created SageMaker Endpoint is ready for prediction requests. The next chapter will explain
+how to make predictions against the Endpoint, how to use different content-types in your requests, and how to extend the Web server
+functionality.
+
+Deploying directly from model artifacts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you already have existing model artifacts, you can skip training and deploy them directly to an endpoint:
+
+.. code:: python
+
+  from sagemaker.tensorflow import TensorFlowModel
+
+  tf_model = TensorFlowModel(model_data='s3://mybucket/model.tar.gz',
+                             role='MySageMakerRole',
+                             entry_point='entry.py',
+                             name='model_name')
+
+  predictor = tf_model.deploy(initial_instance_count=1, instance_type='ml.c4.xlarge')
+
+You can also optionally specify a pip `requirements file <https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format>`_ if you need to install additional packages into the deployed
+runtime environment by including it in your source_dir and specifying it in the ``'SAGEMAKER_REQUIREMENTS'`` env variable:
+
+.. code:: python
+
+  from sagemaker.tensorflow import TensorFlowModel
+
+  tf_model = TensorFlowModel(model_data='s3://mybucket/model.tar.gz',
+                             role='MySageMakerRole',
+                             entry_point='entry.py',
+                             source_dir='my_src', # directory which contains entry_point script and requirements file
+                             name='model_name',
+                             env={'SAGEMAKER_REQUIREMENTS': 'requirements.txt'}) # path relative to source_dir
+
+  predictor = tf_model.deploy(initial_instance_count=1, instance_type='ml.c4.xlarge')
+
+
+Making predictions against a SageMaker Endpoint
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following code adds a prediction request to the previous code example:
+
+.. code:: python
+
+  estimator = TensorFlow(entry_point='tf-train.py', ..., train_instance_count=1,
+                         train_instance_type='ml.c4.xlarge', framework_version='1.10.0')
+
+  estimator.fit(inputs)
+
+  predictor = estimator.deploy(initial_instance_count=1, instance_type='ml.c4.xlarge')
+
+  result = predictor.predict([6.4, 3.2, 4.5, 1.5])
+
+The ``predictor.predict`` method call takes one parameter, the input ``data`` for which you want the SageMaker Endpoint
+to provide inference. ``predict`` will serialize the input data, and send it in as request to the SageMaker Endpoint by
+an ``InvokeEndpoint`` SageMaker operation. ``InvokeEndpoint`` operation requests can be made by ``predictor.predict``, by
+boto3 ``sageMaker.runtime`` client or by AWS CLI.
```

Comment:
this should probably be "SageMaker Runtime client". if you want, could also add a link to https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker-runtime.html

Reply:
ok